### PR TITLE
Setting a default theme depending on a visitor's browser

### PR DESCRIPTION
**Pull request by mikethomashaber**
Added Mobile direct extension and code for setting a default theme depending on a visitor's browser.

This will provide the mobile theme to mobile users.

### CODE REVIEW

File: public_wiki/LocalSettings.php [hunk 1]
```diff
@@ -148,7 +148,17 @@
...
+# Set default theme depending on browser
+# Adding Mobile Detect extension
+wfLoadExtension( 'MobileDetect' );
+$mobile = wfMobileDetect();
+if ( $mobile ) {
+    $wgDefaultSkin = "chick"; # If mobile
+} else {
+    $wgDefaultSkin = "vector"; # If not mobile
+}
+
```
Comment (dpslwk) [inline public_wiki/LocalSettings.php]:
```suggestion
$wgDefaultSkin = 'vector';

# Adding Mobile Detect extension
wfLoadExtension( 'MobileDetect' );

# Update default theme depending on browser
if (wfMobileDetect()) {
    $wgDefaultSkin = 'minerva'; # switch if mobile
}
```

File: public_wiki/LocalSettings.php [hunk 2]
```diff
@@ -334,6 +344,7 @@
 # Adding HMS auth extension
 wfLoadExtension( 'AuthHMS' );
 
+
```
Comment (dpslwk) [inline public_wiki/LocalSettings.php]:
```suggestion
```

**Review by dpslwk — CHANGES_REQUESTED** [review]
See suggestions
and please confirm that the MobileDetect extension files where downloaded for version 1.35 of media wiki not 1.38

File: public_wiki/LocalSettings.php (hunk 1 above)
Comment (mikethomashaber) [reply public_wiki/LocalSettings.php]: I downloaded the version for 1.35, MobileDetect-REL1_35-c137344.tar

Happy for the code suggestion. I'd taken the if/else code from the wikimedia page.

**Comment by mikethomashaber** [comment]
Seems to have broken wiki.nottinghack.org? Unless it's network issues?

**Comment by dpslwk** [comment]
fixed with 4bf758fd224d43891590cea277a4de14e4033263

no idea if its doing its job mind

**Comment by mikethomashaber** [comment]
Ah thanks, that's not the code I saw recommended. Thanks for sorting that. Site is delivering a different skin on my phone, so seems OK.

**Comment by dpslwk** [comment]
No worries

Looked at the diff
https://www.mediawiki.org/w/index.php?title=Extension%3AMobileDetect&type=revision&diff=5359924&oldid=5359921 <https://www.mediawiki.org/w/index.php?title=Extension:MobileDetect&type=revision&diff=5359924&oldid=5359921>

Since its older version and for 1.35 its not been update as the master has
https://github.com/wikimedia/mediawiki-extensions-MobileDetect/commit/7efd1f4397e5b9bc924ddea85396747dd0247c9a

**Comment by mikethomashaber** [comment]
Hi,
Yeah 1.35, the file was



MobileDetect-REL1_35-c137344.tar


ta
Mike

**Pull request merged**